.github/workflows/test-AI.yml

```diff
@@ -0,0 +1,47 @@
+name: Test AI with linting and unit tests
+on: push
+jobs:
+  test-AI:
+    name: AI source linting and testing
+    runs-on: ubuntu-18.04
+    strategy:
+      matrix:
+        python-version:
+          - 3.6
+    steps:
+      - name: Checkout sources
+        uses: actions/checkout@v2
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v1
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Register flake8 annotation problem matcher
+        uses: rbialon/flake8-annotations@v1
+      - name: Set up system dependencies
```

Comment:
It is the longest step. I would recommend to remove it.  Or try to cache it. 

![image](https://user-images.githubusercontent.com/171597/89128558-20303280-d4ff-11ea-93bd-141ce850ea65.png)